src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -41,16 +41,17 @@ internal partial class VisualStudioWorkspaceDiagnosticAnalyzerProviderService :
             // this will allow us to build once and deploy on different versions of VS SxS.
             var vsDteVersion = Version.Parse(dte.Version.Split(' ')[0]); // DTE.Version is in the format of D[D[.D[D]]][ (?+)], so we need to split out the version part and check for uninitialized Major/Minor below
             var assembly = Assembly.Load($"Microsoft.VisualStudio.ExtensionManager, Version={(vsDteVersion.Major == -1 ? 0 : vsDteVersion.Major)}.{(vsDteVersion.Minor == -1 ? 0 : vsDteVersion.Minor)}.0.0, PublicKeyToken=b03f5f7f11d50a3a");
-
-            // Get the analyzer assets for installed VSIX extensions through the VSIX extension manager.
-            var extensionManager = serviceProvider.GetService(assembly.GetType("Microsoft.VisualStudio.ExtensionManager.SVsExtensionManager"));
-            if (extensionManager == null)
+            if (assembly == null)
             {
                 // extension manager can't be null. if it is null, then VS is seriously broken.
```

Comment:
Comment needs revision.

Reply:
Current comment seems as correct as it was before the change.

Reply:
from the bug description, it was talking about VS not running. is there a way to check something like this?

http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/LanguageService/AbstractPackage%25602.cs,175

for this case? so that we don't run solution crawler or servicehub?

Reply:
@heejaechang `GetService` is always allowed to start returning null at an unspecified point at which the IDE has decided to shut down.

Reply:
> @sharwell: Current comment seems roughly as correct as it was before the change.

Not so sure. Before at least the comment matched the "if" block it was within. Now it's referencing things you haven't even checked yet.

Reply:
> GetService is always allowed to start returning null after an unspecified point at which the IDE has decided to shut down.

sure, but I am not asking about the normal shutdown case.

from the bug,

> There are 800-1000 tests in that run (depending on the branch) and without any additional data there's nothing I can do. As far as I know no test in that run launches VS although there are some tests that will invoke devenv.com /update. Is there any chance that running through update end up loading those components? If that ends up being the case that is a valid user scenario.

so, it looks like it is not a normal shutdown where user opens a VS and close it, it is more like test is doing some VS update scenario and we ends up think we are in normal VS start scenario and things shouldn't run get started and at the end, causing crash.

we had similar case for devenv /build mode. so I am just asking whether we can do same check to just avoid the whole thing.

Reply:
➡️ I decided to leave this for fixing as part of #35138 since the build passes and this otherwise mitigates the immediate issue observed in infrastructure. The comment will end up removed.